### ISSUE TITLE
Select and React Testing Library

### ISSUE THREAD

**Issue by Jonur** [mantinedev/mantine#3144]
### What package has an issue

@mantine/core

### Describe the bug

We upgraded Mantine's version from 4.2.10 to 5.9.2 and our form submit unit tests broke and cannot figure out why. Adding debugs we can see the rendered input fields' values update properly. But on submit the values were empty. Then we noticed that the `onChange` functions never fired, which is plain odd, since the values do change.

Firstly, the code:

We have a form which includes three `<Select />` components like so:

```javascript
      <Select
          label={formatMessage({ id: 'vehicle.form.make' })}
          name="make"
          data={makes}
          placeholder={formatMessage({ id: 'vehicle.form.selectMake' })}
          onChange={handleMakeChange}
          searchable
          data-testid="make"
          value={make}
          creatable
          getCreateLabel={(value) => `+ ${value}`}
          onCreate={handleMakeCreate}
        />
```

Both `handleMakeChange` and `handleMakeCreate` just do a couple of state updates (on `useState` generated values).

In the tests, we tried both as:
```javascript
userEvent.click(getByTestId('make'));
userEvent.type(getByTestId('make'), MAKE);
userEvent.keyboard('{enter}');
```

and

```javascript
fireEvent.change(getByTestId('make'), {
        target: {
          value: MAKE,
        },
      });
```

and when we `debug()` the component the generated `<input />` has the right values. Yet, if we want to make any assertions on the `onChange` functions of each of these or if we add `console.log` it does not show. Again, this issue is on the unit tests only. The code runs properly on the browser.

Does anyone know why that is? Or what is the right way trigger these onChange functions?

### What version of @mantine/hooks page do you have in package.json?

5.9.2

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment]
Make sure that you use creatable select correctly, there were breaking changes in 5.0 related to this – https://mantine.dev/core/select/#creatable

**Comment by Jonur** [comment]
@rtivital We did create a version where both onCreate and initial data create/are array of objects `{ value, label: value }`, but still, we get the same result. Have you got an example of a `<Select />` component in TS and React Testing Library that fires change events and you can confirm the onChange function runs?

**Comment by rtivital** [comment]
Does the component work as expected in the browser?

**Comment by Jonur** [comment]
Yes @rtivital it works as expected on the browser. It only the unit tests that stopped working after the version update.

Have you got an example of a `<Select />` component in TS and React Testing Library that fires change events and you can confirm the onChange function runs?

**Comment by rtivital** [comment]
No, we do not have such tests, I was planning to add them in of the next 5.x releases

**Comment by Jonur** [comment]
That's alright, thanks @rtivital . Let's hope somebody else might have. 😊

**Comment by aiheon** [comment]
If you're not on the latest version of UserEvent, I would advise to bump, it's a lot easier to work with version 14.X.X 😄 


```
it('call spy on', async () => {
    const spy = jest.fn(); // of _.noop() or another mock call comming from your framework of choice :p

    render(<MyComponentWithASelect onChange={spy} />);

    const mantineSelect = screen.getByRole('searchbox', {name: /make/i});

    // if you have a default value, clear the input first
    // await userEvent.clear(mantineSelect);

    await userEvent.type(mantineSelect, 'so patate');

    expect(mantineSelect).toHaveValue('so patate'); //
    expect(spy).toHaveBeenCalled();
});
```


I cannot guarantee that the expect of the spy will work though, i'm using form providers to handle changes so `onChange` is "under the cover" in my code 🙈

![image](https://user-images.githubusercontent.com/63734941/206706409-0a90e9e6-54bb-4bcf-beb9-fb3635391ad8.png)
